x/mint/types/params.go

```diff
@@ -41,7 +38,13 @@ func (p Params) Validate() error {
 	if err := validateMintDenom(p.MintDenom); err != nil {
 		return err
 	}
-	if err := validateAnnualReductionFactor(p.AnnualReductionFactor); err != nil {
+	if err := validateInflationCoef(p.InflationCoef); err != nil {
+		return err
+	}
+	if err := validateBoundingAdjustment(p.BoundingAdjustment); err != nil {
+		return err
+	}
+	if err := validateTargetBoundingRatio(p.TargetBoundingRatio); err != nil {
```

Comment:
I think those checks are not enough, and furthermore those parameters should be checked together IMHO, here is why:

They intervene in the `c - b / bt` bonding ratio coef of the inflation rate, which must be positive in order to prevent a possible negative inflation rate.

To prevent this term to be negative:
`c - b / bt ≥ 0` => `c ≥ b / bt`

And assuming `b` goes towards `1`
`c ≥ 1 / bt` => `c • bt ≥ 1`

Reply:
Thanks, good catch ! 

So we need to check that the result of `b / bt` must be greater or equal to `c`. Given `b ∈ [0;1]`, we can assume that `bt ∈ ]0; 1]` but we need to check `bt ≥ 1 / c` to avoid the result of `b / bt` be greater or equal than `c` and avoid negative result. 

Are you agree with that ? If yes I will add this check in the validate function.